src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepository.java

```diff
@@ -443,10 +443,10 @@ public void postFinishedComment(
    *
    * @param pullRequestId pull request ID
    * @param version pull request version
-   * @return true if the merge succeeds, false if the server reports an error
+   * @return "MERGED" if the merge succeeds, server response otherwise
```

Comment:
I don't like using String to return what should effectively be different types.
We could add a proper DTO here or we could use an Optional return type -- empty on success and containing the response on failure

Reply:
sine we already have StashPullRequestResponseValue can we return that instead of a String?

Reply:
The error message in a different format. If we try to parse it as StashPullRequestResponseValue, we would lose important information. The caller is more interested in the error message than in the successfully merged PR (as long as the success is indicated), so I guess Optional with String would work.